torch/testing/_internal/common_methods_invocations.py

```diff
@@ -955,7 +955,36 @@ def sample_inputs_diff(op_info, device, dtype, requires_grad, **kwargs):
 
     return tuple(sample_inputs)
 
-def sample_inputs_index_select(op_info, device, dtype, requires_grad, **kwargs):
+def sample_inputs_gradient(op_info, device, dtype, requires_grad):
+    sample_inputs = []
+
+    test_cases_float = (
+        ((S,), None, None),
+        ((S,), 2.0, None),
+        ((S, S), None, None),
+        ((S, S), [2.0, 2.1], None),
+        ((S, S), [2.0, 2.1], (0, 1)),
+        ((4, 4, 4), [2., 1.], (0, 1)),
+    )
+    for size, spacing, dim in test_cases_float:
+        t = make_tensor(size, device, dtype, low=None, high=None, requires_grad=requires_grad)
+        sample_inputs.append(SampleInput(t, kwargs=dict(dim=dim, domain=spacing)))
+
+    test_cases_tensor = (
+        ((3, 3, 3), [(1.0, 2.0, 3.0), (4.0, 5.0, 6.0)], (0, -1)),
+        ((3, 3, 3), [(1.0, 3.0, 2.0), (8.0, 6.0, 1.0)], (0, 1)),
+    )
+    for size, coordinates, dim in test_cases_tensor:
+        t = make_tensor(size, device, dtype, low=None, high=None, requires_grad=requires_grad)
+        coordinates_tensor_list = []
+        for coords in coordinates:
+            a = torch.tensor(coords, dtype=torch.float64, device=device)
```

Comment:
What about putting coords in the same dtype as input?

Reply:
If we want to allow everything that sounds great, but we should extend cross dtype testing here and in the main test

Reply:
Suggestion: fixture needs to be extended with integer and complex values, then instantiation can be updated to:

```
for coords, dtype in product(coordinates, (torch.long, torch.float64, torch.complex128)):
  a = torch.tensor(coords, dtype=dtype, device=device)
```

but you'll need to find a way to filter coords to the appropriate dtype or convert the values in coords to the appropriate dtype.